common/service/config/ringpop.go

```diff
@@ -202,6 +209,42 @@ func (factory *RingpopFactory) getChannel(dispatcher *yarpc.Dispatcher) (*tcg.Ch
 	return ch, nil
 }
 
+type minimalResolver interface {
+	LookupHost(ctx context.Context, host string) (addrs []string, err error)
+}
+
+type dnsProvider struct {
+	Resolver        minimalResolver
+	UnresolvedHosts []string
```

Comment:
can you make this unresolved a set ? that way, we can also be sure that there are no duplicates in the input list and so, no duplicates on the output list of resolved addrs. The population of set can be done in a constructor of form newDNSProvider(hosts []string, resolver dnsHostResolver) *dnsProvider {}